src/axom/inlet/inlet_utils.cpp

```diff
@@ -165,5 +189,33 @@ void markAsStructCollection(axom::sidre::Group& target)
   }
 }
 
+void markRetrievalStatus(axom::sidre::Group& target, const ReaderResult result)
+{
+  if(result != ReaderResult::Success)
+  {
+    target.createViewScalar("retrieval_status", static_cast<int>(result));
+  }
+}
+
+ReaderResult collectionRetrievalResult(const bool homogeneous, const bool empty)
+{
+  // First check if the collection was entirely the wrong type
+  if(!homogeneous && empty)

```

Comment:
!empty?

Reply:
The "empty" refers to the C++ container used to store the requested elements, which will be empty if none were present.  The name is a little confusing since it could also refer to the structure in the input file, so I've renamed it accordingly.